delegator/engine.go

```diff
@@ -91,16 +98,33 @@ func (e *Engine) Run(ctx context.Context, opts config.Delegator) error {
 		middleware.Recoverer,
 	)
 
-	// register endpoints
+	// register regular HTTP endpoints
 	r.HandleFunc("/status", e.Status)
 	r.Route(fmt.Sprintf("/network/{%s}", keyNetwork), func(r chi.Router) {
-		r.Use(e.NetworkContext)
+		r.Use(e.NetworkPathContext)
 		r.HandleFunc("/status", e.NetworkStatus)
 		r.Route(fmt.Sprintf("/{%s}", keyFeature), func(r chi.Router) {
 			r.HandleFunc("/*", e.Redirect)
 		})
 	})
 
+	// handle subdomain-based routing
+	if e.domain != "" {
```

Comment:
This part I'm a little confused about, shouldn't we always want to spin this up? Or is this instantiated individually for each IPFS network that gets registered?

Reply:
Well if you don’t provide a domain it’ll match for “name.api.” Which is not useful, and potentially have all sorts of terrible side effects

Reply:
Oh okay I guess I was confused then, this would be the `domain` that nexus is running behind? ie `nexus.temporal.cloud` so this would then match `*.nexus.temporal.cloud` yea?

Reply:
The next 6 lines show what it matches, but in config you would put nexus.temporal.cloud